### DISCUSSION TITLE
Support for Inline Images

### DISCUSSION THREAD

**Discussion by madc** [formfcw/directus-extension-flexible-editor#43] [Feature Request]
I'm aware this is somewhat possible using the M2A feature, but also think the editor should be able to handle inline images directly.
TipTap offers an extension for rendering (not uploading) images: [Images](https://tiptap.dev/docs/editor/api/nodes/image).

**Functionality**:

- Add Image from file manager
- Directly upload Image
- Re-Select/Remove Image
- Edit Image Properties (Title, Description,..)

This feature could be inspired by the way, [the EditorJS extension](https://github.com/dimitrov-adrian/directus-extension-editorjs-interface) handles images.

Select Image Button:
![image](https://github.com/formfcw/directus-extension-flexible-editor/assets/343392/1bccafcb-0317-4284-a9d5-d0fc3162447f)

Select Image Screen:
![image](https://github.com/formfcw/directus-extension-flexible-editor/assets/343392/bb1758a9-0d70-4cf1-9304-ace97dc2f461)

With selected Image:
![image](https://github.com/formfcw/directus-extension-flexible-editor/assets/343392/ca0d0ca0-708a-4265-95f7-ad5f6fdbd36d)

Some Options:
![image](https://github.com/formfcw/directus-extension-flexible-editor/assets/343392/001e309b-2783-46d4-b574-1e578a7f74bf)

EditorJS stores some information about the image (i.e. the id) directly in the json object and does not care about the collection afterwards. So there is loads of room for improvement, but I feel that is is possible.

**Comment by Xinnerpeace** [comment]
Any solution for this most up-voted request?

**Reply by formfcw** [reply]
Wanna contribute?
